### DISCUSSION TITLE
python ide setup using lazyvim

### DISCUSSION THREAD

**Discussion by mohit2152sharma** [LazyVim/LazyVim#907] [Q&A] [answered]
Hi,
I am fairly new to lazyvim (in fact neovim too) and I am trying to setup a IDE type of thing for python. By IDE type of thing I mean: setting up debugger, setting up autocompletion (or intellisense) and autoformatting using black. And I am struggling with all of the above. If someone has setup the above things for python, could you please share your lazyvim setup.

I have read through, the documentations and for a beginner I found them a bit confusing. For example, If I go to [nvim-dap](https://www.lazyvim.org/plugins/extras/dap.core), I understood that I need to create a `nvim-dap.lua` file inside `~/.config/nvim/lua/plugins/` but I couldn't understand if I need to put full spec or only `opts` in there. I tried with `full-spec` and I get the following window when i go into `dap-ui`.

<img width="953" alt="Screenshot 2023-06-08 at 10 56 57 PM" src="https://github.com/LazyVim/LazyVim/assets/26904579/ed769f44-3eb7-4141-a024-d650b6e244ab">

**Comment by dpetka2001** [comment]
You can take a look at mine [here](https://github.com/dpetka2001/dotfiles/blob/main/dot_config/nvim/lua/plugins/extras/lang/python.lua).

**Reply by mohit2152sharma** [reply]
Thanks a lot @dpetka2001 , I just wanted to confirm If I am getting it right, I am only interested in python stuff, so for that I only need to setup `plugins/extra/lang/python.lua` or do I need to add any other file?

**Reply by dpetka2001** [reply]
No just that file. Just put the file in `.config/nvim/lua/plugins`. If you follow my structure `.config/nvim/lua/plugins/extras/lang/python.lua`, then you need to import it somewhere(preferably in `lua/config/lazy.lua`). So just before `{ import = "plugins" }`, put `{ import = "plugins.extras.lang" }`. But if this seems confusing, just put the `python.lua` file in `.config/nvim/lua/plugins` and you're done.

**Reply by andyjia** [reply]
@dpetka2001 
Thank you. I put your python.lua file in .config/nvim/lua/plugins, then I restarted nvim, but the error happened, what I missed?

```
Failed to run `config` for mason-nvim-dap.nvim
vim/shared.lua:0: dst: expected table, got nil
# stacktrace:
  - vim/shared.lua:0 _in_ **validate**
  - vim/shared.lua:0 _in_ **list_extend**
  - .config/nvim/lua/plugins/python.lua:36 _in_ **values**
  - .config/nvim/lua/config/lazy.lua:9
  - .config/nvim/init.lua:2
```

**Reply by andyjia** [reply]
@dpetka2001 
I missed this to lazy.nvim. Thank you.
`{ import = "lazyvim.plugins.extras.dap.core" },`

**Comment by fredrikaverpil** [comment, marked as answer]
@mohit2152sharma I'm working on a daily basis in Python with neotest/dap/formatting/linting and LSP.

I'm not using the LazyVim-provided Python extras as I got setup prior to that existing. You can have a look at my setup [here](https://github.com/fredrikaverpil/dotfiles/blob/main/nvim-lazyvim/lua/plugins). I'm using null-ls for now, so take note of how some plugins are disabled (not active) in e.g. `format.lua` and `lint.lua`.

**Reply by mohit2152sharma** [reply]
Thank you @fredrikaverpil for sharing your setup. There are some interesting plugins in your setup, which I didn't know about. Thank you again for sharing.
